calico/felix/test/test_fip.py

```diff
@@ -0,0 +1,64 @@
+import collections
+import mock
+
+from calico.felix.test.base import BaseTestCase
+from calico.felix.fipmanager import FloatingIPManager
+from calico.felix.frules import CHAIN_FIP_DNAT, CHAIN_FIP_SNAT
+
+Config = collections.namedtuple('Config', ['IFACE_PREFIX', 'METADATA_IP', 'METADATA_PORT'])
+
+
+class TestFloatingIPs(BaseTestCase):
+    def setUp(self):
+        super(TestFloatingIPs, self).setUp()
+        self.iptables_updater = mock.MagicMock()
+        self.config = Config('tap', None, 8775)
+
+    def getFloatingIPManager(self, ip_version):
+        return FloatingIPManager(
+            config=self.config,
+            ip_version=ip_version,
+            iptables_updater=self.iptables_updater
+        )
+
+    def assert_iptables_update(self, args, dnat_chain, snat_chain):
+        self.assertItemsEqual(args[0][0][CHAIN_FIP_DNAT], dnat_chain)
+        self.assertItemsEqual(args[0][0][CHAIN_FIP_SNAT], snat_chain)
+        # We shouldn't have any dependency chains.
+        self.assertEqual(args[0][1], {})
+
+    def _run_fip_manager_loop(self, ip_version, maps, expected_dnat, expected_snat):
+        self.config = Config('tap', '127.0.0.1', 8775)
+
+        f = self.getFloatingIPManager(ip_version)
+        f.update_endpoint('endpoint_id', maps, async=True)
+        self.step_actor(f)
+
+        self.iptables_updater.assertCalledOnce()
```

Comment:
I don't think assertCalledOnce exists.  Sadly the Mock doesn't fail if you call a non-existent method.  I've started using this pattern, which is more verbose but avoids this issue:
```
self.assertEqual(mock.mock_calls, [mock.call(<args>), mock.call(<args>)])
```